source/client/RunManagement.py

```diff
@@ -0,0 +1,250 @@
+from common.Card import Card
+
+"""This file contains methods used in processing runs for Liverpool.
+
+Some Liverpool specific stuff:
+it assumes that Aces can be Hi or Low but not both (K,A,2 not allowed).
+Enforces rule that you must have two natural cards between wilds in a run.
+
+In order to preserve backwards compatibility with June 2020 distribution, it assumes that card.tempnumber is
+not passed back and forth between the client and server. This only causes ambiguity in wilds and 
+Aces at the ends of runs.   Method restoreRunAssignment takes care of this by assigning
+wilds and Aces on ends of runs the appropriate tempnumber before processRuns is called.
+To preserve info on whether Ace is assigned hi or low, if Ace is assigned low, then tempnumber is set to -1.
+"""
+
+def processRuns(card_group, wild_numbers):
+    """ handle sorting of run, including placement of wilds.  Handles minor rule checking.
+
+    # processRuns does not presume length requirement or that all cards are in same suit.
+    # it DOES presume that if Aces are not wild, then they are hi or low, but not both.
+    # IF ACE CAN BE HIGH OR LOW (very unusual) THAN AUTOMATICALLY MAKING IT LOW.
+    """
+    # print("in RunManagement, processRuns")
+    # print(card_group)
+    card_group.sort(key=lambda wc: wc.tempnumber)
+    # print(card_group)
+    first_card = True
+    groups_wilds = []
+    temp_run_group = []
+    aces_list =[]
+    gap_flag = False
+    for card in card_group:                    # separate unassigned wilds and Aces from rest of run.
+        # pull out wilds that have not been assigned (do this before Aces, in case rules changed so that Aces are wild).
+        if card.tempnumber in wild_numbers:
+            groups_wilds.append(card)
+            # pull out Aces that have not been assigned hi or low, but don't pull out Wilds assigned to low Ace slot.
+        elif card.tempnumber == 1 and card.number == 1:
+            aces_list.append(card)
+        else:
+            temp_run_group.append(card)
+    card_group  = []                         # rebuild card_group below
+    for card in temp_run_group:
+        if first_card:
+            first_card = False
+            card_group.append(card)
+        else:
+            abs_temp_number = abs(card_group[-1].tempnumber)  # Aces designated to be low have tempnumber = -1.
+            if card.tempnumber == (abs_temp_number + 1):
+                card_group.append(card)
```

Comment:
This assumes the cards are sorted by number - which I am sure is true but should go in your assumptions at the top.